cocotb/triggers.py

```diff
@@ -469,7 +482,37 @@ def __str__(self):
 
 
 class Join(with_metaclass(_ParameterizedSingletonAndABC, PythonTrigger)):
-    """Join a coroutine, firing when it exits."""
+    r"""Fires when a :func:`fork`\ ed coroutine completes
```

Comment:
For such cases, Sphinx has "You may supply an explicit title and reference target, like in reST direct hyperlinks: :role:`title <target>` will refer to target, but the link text will be title." 
(https://www.sphinx-doc.org/en/1.5/markup/inline.html#cross-referencing-syntax)